plugin/action/mask/mask.go

```diff
@@ -0,0 +1,158 @@
+package mask
+
+import (
+	"regexp"
+
+	"github.com/ozonru/file.d/fd"
+	"github.com/ozonru/file.d/pipeline"
+	insaneJSON "github.com/vitkovskii/insane-json"
+	"go.uber.org/zap"
+)
+
+/*{ introduction
+Remask plugin matches event with regular expression, and substitution successfully matched symbols via template.
+You can set many expressions.
+
+
+**Example:**
+```yaml
+pipelines:
+  example_pipeline:
+    ...
+    actions:
+    - type: mask
+      masks:
+        - mask:
+			re2: "\b\d{1,4}\D?\d{1,4}\D?\d{1,4}\D?\d{1,4}\b"
+          	mask: "****-****-****-****"
+    ...
+```
+There are three variants of math&replace methods:
+* Call regex.Replace for every string in event (maskAll)
+* First call regex.Match, if matching successful then call regex.Replace (maskIfMatched)
+* First call regex.FindMatchingIndex, then replace finded symbol by index(maskByIndex)
+
+}*/
+
+type Plugin struct {
+	config *Config
+	re     []*regexp.Regexp
+	logger *zap.SugaredLogger
+}
+
+//! config-params
+//^ config-params
+type Config struct {
+	//> @3@4@5@6
+	//>
+	//> List of Masks
+	Masks []Mask `json:"masks"` //*
+}
+
+//! config-params
+//^ config-params
+type Mask struct {
+	//> @3@4@5@6
+	//>
+	//> Regular expression used for masking
+	Re string `json:"re2" default:"" required:"true"`
+
+	//> @3@4@5@6
+	//>
+	//>	Substitution mask
+	Substitution string `json:"substitution" default:"" required:"true"`
+}
+
+func init() {
+	fd.DefaultPluginRegistry.RegisterAction(&pipeline.PluginStaticInfo{
+		Type:    "mask",
+		Factory: factory,
+	})
+}
+
+func factory() (pipeline.AnyPlugin, pipeline.AnyConfig) {
+	return &Plugin{}, &Config{}
+}
+
+func (p *Plugin) Start(config pipeline.AnyConfig, params *pipeline.ActionPluginParams) {
+	if params != nil && params.Logger != nil {
+		p.logger = params.Logger
+	}
+	p.config = config.(*Config)
+	p.re = make([]*regexp.Regexp, len(p.config.Masks))
+	for i, mask := range p.config.Masks {
+		p.logger.Infof("compiling regexp %s/ - %s", mask.Re, mask.Substitution)
+		p.re[i] = regexp.MustCompile(mask.Re)
+	}
+}
+
+func (p *Plugin) Stop() {
+}
+
+func (p *Plugin) maskByIndex(value string) (*string, bool) {
```

Comment:
here you don't need a pointer to a string, also you should return []byte, as insane-json has fast function to deal with bytes